Worked on "All Groups" Page

alliedrake opened the pull request.

alliedrake committed:
Worked on "All Groups" Page

alliedrake committed:
Added more content/customization to all group page

ParkerNix commented on payback/src/pages/allGroups/index.jsx:
```diff
@@ -7,27 +7,36 @@ export const allGroups = () => {
-                    <div className ='groupsCardHeart'>
-                        {/* <img src="images/groupsCardHeart.png" alt="heart"> */}
+            <div className='allGroupsPageHeader'>  
+                <h4>All Groups</h4>
```
So with headers, it's better for accessibility if you use headings in numerical order, going from h1 to h2, and so on. If you still want to use the style for a different heading (like this h4), you can use an h1 tag and then give it `className="h4"`. This is a bootstrap class, so it has styling that it will automatically apply to the element. Feel free to change the class' style in the index.css file to adjust it how you want.

[Bootstrap typography](https://www.w3schools.com/bootstrap5/bootstrap_typography.php)
[A11Y heading checklist](https://www.a11yproject.com/checklist/#headings)

ParkerNix commented on payback/src/pages/allGroups/index.jsx:
```diff
@@ -7,27 +7,36 @@ export const allGroups = () => {
+            </div>
+        
+
+            <div className ='groupsContainer'>
```
Here's another place to consider using bootstrap instead. So with this, you can do `className="container"` instead and it will give you a container that has  the centering code already applied. That should help so that you aren't constantly going in and typing in styles for it. I would also consider wrapping the whole page and then using a bootstrap row here as well as for the section where you currently have the "All Groups" heading. It might help to look at how I have it with the newExpense page.

[bootstrap containers](https://www.w3schools.com/bootstrap5/bootstrap_containers.php)
[bootstrap grid, has stuff about row](https://www.w3schools.com/bootstrap5/bootstrap_grid_system.php)

ParkerNix commented on payback/src/pages/allGroups/index.jsx:
```diff
@@ -7,27 +7,36 @@ export const allGroups = () => {
+        
+
+            <div className ='groupsContainer'>
+                <div className ='groupsCard'>
```
There's also a bootstrap card I think could be good here. I'd give it a try and see if you can use it, it would help cut down on the length of the CSS file.

[bootstrap cards](https://www.w3schools.com/bootstrap5/bootstrap_cards.php)

ParkerNix commented on payback/src/pages/allGroups/index.jsx:
```diff
@@ -7,27 +7,36 @@ export const allGroups = () => {
         </div>
 
-        <h4>All Groups</h4>
+        <div className='allGroupsPage'>
```
I would drop this div, unless you had any plans for it.

ParkerNix commented on payback/src/index.css:
```diff
@@ -168,13 +176,50 @@ button {
\ No newline at end of file
+}
+
+.groupsCardButton button {
```
I would try to make this class name a bit more general since it's something we can apply to many buttons. Actually, since most of the buttons are the same, we can probably use the bootstrap button instead and customize it with what you have here. That way, if there's a bigger button (like maybe the one from the modals) we can use these styles and then just adjust it as needed with a different class.

ParkerNix commented on payback/src/index.css:
```diff
@@ -168,13 +176,50 @@ button {
+flex-grow: 0;
+display: flex;
+flex-direction: row;
+justify-content: center;
```
A few of these styles might need some touch-ups, but I'll save that for after we fix what I have commented already. I can't see what the styles look like yet.

ParkerNix commented on payback/src/pages/allGroups/index.jsx:
```diff
@@ -7,27 +7,36 @@ export const allGroups = () => {
-                        {/* <img src="">
-                        <img src="">
-                        <img src=""> */}
+                    <div className ='groupsCardRowTwo'>
```
Might be another place for a bootstrap row, you might also consider bootstrap flex instead. It should do everything a normal flexbox would do, which I think would work better here. check it out though!

[bootstrap flex](https://www.w3schools.com/bootstrap5/bootstrap_flex.php)

ParkerNix commented on payback/src/index.css:
```diff
@@ -11,3 +11,170 @@ code {
+/* 
+:root {
+  --main-bg-color: #FDFAFF;
+  --h1-color: #3d008a;
```
Thanks for doing these! You might want to consolidate these into one color instead of repeating them like this. Maybe just have one variable called header-color-d-purple or sumn, something siliar for the others if you can.

ParkerNix commented on payback/src/index.css:
```diff
@@ -11,3 +11,170 @@ code {
+}
+
+
+@font-face {
```
Shouldn't need to define these, adobe fonts has an API that loads them in so we can save some space.

ParkerNix commented on payback/src/index.css:
```diff
@@ -11,3 +11,170 @@ code {
+}
+
+h2 {
+  color: var(--h2-color);
```
perhaps consolidate these as well so that the elements that have the same color are defined together

ParkerNix commented on payback/src/index.css:
```diff
@@ -11,3 +11,170 @@ code {
+  font-size: 14px;
+}
+
+button {
```
can you just add the class for a bootstrap button so these styles get applied to it? There are a few places where I might ask you to use and a tag instead and then give it the bootstrap button class so it looks like a button. This can help with accessibility as well, specifically if it's a link that moves you to another page rather than a button that does something on the same page.

alliedrake committed:
Consolidated some root customizations

alliedrake committed:
Made create group page

Tried coding It based off of your suggestions but definitely will need to check in with you about class names and some more bootstrap stuff, I just coded out the structure of the page for the time being

alliedrake committed:
Update index.css

Consolidated properties

alliedrake committed:
Update index.jsx

Saw your suggestions and will check in with you on Monday

alliedrake committed:
Merge pull request #1 from ParkerNix/main

Style, fonts

alliedrake committed:
Added asset folder

Added within the components folder
